usm_rest_api/v1/views/utils.py

```diff
@@ -0,0 +1,267 @@
+import random
+import logging
+import uuid
+import time
+
+from usm_rest_api.models import Cluster
+from usm_rest_api.v1.serializers.serializers import ClusterSerializer
+from usm_rest_api.models import Host
+from usm_rest_api.v1.serializers.serializers import HostSerializer
+
+from usm_wrappers import salt_wrapper
+from usm_wrappers import utils as usm_wrapper_utils
+
+
+log = logging.getLogger('django.request')
+
+
+CLUSTER_TYPE_GLUSTER = 1
+CLUSTER_TYPE_CEPH = 2
+STORAGE_TYPE_BLOCK = 1
+STORAGE_TYPE_FILE = 2
+STORAGE_TYPE_OBJECT = 3
+STATUS_INACTIVE = 1
+STATUS_ACTIVE_NOT_AVAILABLE = 2
+STATUS_ACTIVE_AND_AVAILABLE = 3
+HOST_TYPE_MONITOR = 1
+HOST_TYPE_OSD = 2
+HOST_TYPE_MIXED = 3
+HOST_TYPE_GLUSTER = 4
+HOST_STATUS_INACTIVE = 1
+HOST_STATUS_ACTIVE = 2
+ACCEPT_MINION_TIMEOUT = 3
+CONFIG_PUSH_TIMEOUT = 15
+
+
+def add_gluster_host(hostlist, newNode):
+    if hostlist:
+        host = random.choice(hostlist)
+
+        rc = salt_wrapper.peer(host, newNode)
+        if rc is True:
+            return rc
+        #
+        # random host is not able to peer probe
+        # Now try to iterate through the list of hosts
+        # in the cluster to peer probe until
+        # the peer probe is successful
+        #
+
+        # No need to send it to host which we already tried
+        hostlist.remove(host)
+        for host in hostlist:
+            rc = salt_wrapper.peer(host, newNode)
+            if rc is True:
+                return rc
+        return rc
+    else:
+        return True
+
+
+def create_cluster(cluster_data):
+    # create the cluster
+    cluster_data['cluster_id'] = str(uuid.uuid4())
+    try:
+        clusterSerilaizer = ClusterSerializer(data=cluster_data)
+        if clusterSerilaizer.is_valid():
+            clusterSerilaizer.save()
+        else:
+                log.error(
+                    "Cluster Creation failed. Invalid clusterSerilaizer")
+                log.error(
+                    "clusterSerilaizer Err: %s" % clusterSerilaizer.errors)
+                raise Exception(
+                    "Cluster Creation failed. Invalid clusterSerilaizer",
+                    clusterSerilaizer.errors)
+    except Exception, e:
+        log.exception(e)
+        raise e
+
+
+def setup_transport_and_update_db(cluster_data, nodelist):
+    log.debug("Inside setup_transport_and_update_db function")
+    failedNodes = []
+
+    # Create the Nodes in the cluster
+    # Setup the host communication channel
+    # current_task.update_state(state='ESTABLISH_HOST_COMMUNICATION')
```

Comment:
yes, this is taken care from the calling task. I will remove the commented lines